### ISSUE TITLE
AutoComplete endpoint in Search provides response in (almost) random order

### ISSUE THREAD

**Issue by johnataylor** [NuGet/NuGetGallery#2437]
Given a large number of potential matches (based on ngrams) the query attempts to return results ordered by relevance. However, at least initially, for the first few characters typed, because this is ngram matches the only thing the relevance is judged by is essentially downloads. And downloads only influence the top 250. Otherwise the order is determined by other factors, such as the time the package was actually added to the index.

This is complicated further by the fact that the ngram tokenization is applied after the regular camel/dot split tokenization we do on the id field. The impact of this is subwords not at the front of the Id can match the query.

In other words the final result is that the order is essentially random.

The fix here is to decide on a reasonable order, including a definition of relevance and return that.

This fix is purely an algorithm change internally with the server. It does not impact the contract with the client, just the data that is returned.

**Comment by joelverhagen** [comment]
This has greatly changes on Azure Search. We now use the `*` (prefix) operator and depend on the default scoring profile which is essentially by download count.
